app/server.py

```diff
@@ -22,6 +22,504 @@ def allRecords():
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+import hashlib  # Hashing module
+import time # Time module
+import jwt  # JSON Web Token module
+
+# Globals
+db_users = client.userdb
+ISSUER = "reconFIU_server"
+EXP_TIME = 86400 # 1 day
+SECRET_KEY = "RECON_FIU_CEN_4010"  # TODO: Should be Randomized and sent to Client Browser
+
+# Show all documents in userdb
+@app.route('/getallusers')
+def allUsers():
+    cursor = db_users.users.find().limit(100)
+    results = []
+    for document in cursor:
+        results.append({'username': document['username'], 'password': document['password'], 'token': document['token'] })
+    return jsonify({ 'count': cursor.count() , 'results': results})
+
+
+# Find a user in userdb
+@app.route('/finduser/<username>')
+def find_user(username):
+    user = db_users.users.find_one({'username': username})
+    if user:
+        results = []
+        results.append({'username': user['username'], 'password': user['password'], 'token': user['token'] })
+        return jsonify(results)
+    else:  # User not found
+        return "Error: User: " + username + " not found"
+
+
+# Create a user document in userdb
+@app.route('/adduser/<username>/<password>')
+def add_user(username, password):
+    user = db_users.users.find_one({'username': username})
+    if user:
+        return "Error: User: " + username + " already exists"
+    else:  # User not found
+        hash_password = hash_pass(password)
+        new_user = {'username': username, 'password': hash_password, 'token': ""}
+        # Add to userdb
+        db_users.users.insert(new_user)
+        return "User: " + username + " has been created with Pass: " + hash_password
+
+
+# Login user, authorization
+@app.route('/login/<username>/<password>')
+def login(username, password):
+    user = db_users.users.find_one({'username': username})
+    if user:
+        if user['token'] == '':
+            # Determine if password is correct
+            if hash_pass(password) == user['password']:
+                token = generate_token(username)
+                print token
+                # TODO: Send Token Here
+                # Store the token in userdb
+                db_users.users.update_one({'username': username}, {'$set': {'token': token}})
+                return "User: " + username + " has logged in"
+            else:
+                return "Error: Given password does not match with user: " + username
+        else:
+            if is_valid_token(user['token'], user['username']):
+                return "Error: User: " + username + " is already logged in"
+            else:
+                # Reset the token in userdb
+                db_users.users.update_one({'username' : username}, { '$set': {'token': ""} })
+                return "Error: Invalid token received, reset token"
+    else:  # User not found
+        return "Error: User: " + username + " not found"
+
+
+# Logout user, authorization
+# TODO: Instead of Username, receive Token
```

Comment:
For the todo if you use a post you can which one was passed. wether `user` and `password` or token